src/redis-benchmark.c

```diff
@@ -628,16 +635,11 @@ static void writeHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
         if (config.cluster_mode && c->staglen > 0) setClusterKeyHashTag(c);
         atomicGet(config.slots_last_update, c->slots_last_update);
         if (config.qps) {
-            pthread_mutex_lock(&request_mutex);
-            if (config.requests_available <= 0) {
-                pthread_mutex_unlock(&request_mutex);
-                return;
-            } else {
-                config.requests_available -= 1;
-                pthread_mutex_unlock(&request_mutex);
-            }
+            long long timestamp = popItem();
+            if (timestamp == -1) return;
+            else  c->start = timestamp;
```

Comment:
I think there's a problem here. This timestamp is not the actual timestamp sent to the Server.